internal/testrunner/runners/system/runner.go

```diff
@@ -228,11 +229,85 @@ func createTestRunID() string {
 	return fmt.Sprintf("%d", rand.Intn(testRunMaxID-testRunMinID)+testRunMinID)
 }
 
-func (r *runner) getDocs(dataStream string) ([]common.MapStr, error) {
+func buildAllFieldsBody() io.Reader {
+	var b strings.Builder
+	b.WriteString("\n")
+	b.WriteString(`{
+	  "fields": ["*"]
+	}`)
+	b.WriteString("\n")
+	return strings.NewReader(b.String())
+}
+
+func (r *runner) isSyntheticsEnabled(dataStream, componentTemplatePackage string) (bool, error) {
+	logger.Debugf("check whether or not synthetics is enabled (component template %s)...", componentTemplatePackage)
+	resp, err := r.options.API.Cluster.GetComponentTemplate(
+		r.options.API.Cluster.GetComponentTemplate.WithName(componentTemplatePackage),
+	)
+	if err != nil {
+		return false, fmt.Errorf("could not get component template from data stream %s: %w", dataStream, err)
+	}
+	defer resp.Body.Close()
+
+	var results struct {
+		ComponentTemplates []struct {
+			Name              string `json:"name"`
+			ComponentTemplate struct {
+				Template struct {
+					Mappings struct {
+						Source *struct {
+							Mode string `json:"mode"`
+						} `json:"_source,omitempty"`
+					} `json:"mappings"`
+				} `json:"template"`
+			} `json:"component_template"`
+		} `json:"component_templates"`
+	}
+
+	if err := json.NewDecoder(resp.Body).Decode(&results); err != nil {
+		return false, fmt.Errorf("could not decode search results response: %w", err)
+	}
+
+	if len(results.ComponentTemplates) == 0 {
+		logger.Debugf("no component template found for data stream %s", dataStream)
+		return false, nil
+	}
```

Comment:
In current elastic stack version, there is no component template for `sql_input` package (input) for the Elastic stack version started (8.6.2).
 
In that scenario, I assume that `synthetic` is not enabled.